addons/event_sale/tests/test_event_sale_with_pricelist.py

```diff
@@ -0,0 +1,133 @@
+    configuration of the price list.
+    Note that as this mechanism is duplicated in product and product template, we test both of them.
+    """
+    @classmethod
+    def setUpClass(cls):
+        super(TestEventSaleTicketWithPriceList, cls).setUpClass()
+        cls.ticket_price = 100.0
+        cls.event_ticket = cls.env['event.event.ticket'].create({
+            'name': 'Event ticket',
+            'price': cls.ticket_price,
+            'event_id': cls.event_0.id,
+            'product_id': cls.event_product.product_variant_id.id,
+        })
+
+    def create_sale_order(self, qty):
+        """Create a sale order for qty of the event_ticket.
+
+        :param int qty: number of ticket ordered
+        :return: (sale_order, sale_order_line)
+        """
+        so = self.env['sale.order'].create({
+            'partner_id': self.env.user.partner_id.id,
+            'pricelist_id': self.price_list.id,
+            'date_order': self.now,
+        })
+        sol = self.env['sale.order.line'].create({
+            'product_id': self.product_or_template.product_variant_id.id,
+            'order_id': so.id,
+            'event_id': self.event_0.id,
+            'event_ticket_id': self.event_ticket.id,
+            'product_uom_qty': qty,
+        })
+        return so, sol
+
+    def assert_amount_total(self, actual_sale_order, expected_amount_total, qty, msg):
+        """ Assert that the sale order amount total and the ticket price reduce tax incl have the expected value.
+
+        The goal is to test the 2 ways of computing the price (based on the price list): through sale order and
+        through ticket.
+
+        Assert that actual_sale_order.amount_total = expected_amount_total
+        Assert that amount total computed with ticket price reduce tax incl = expected_amount_total
+        """
+        order_qty_msg = f'(qty: {qty})'
+        self.assertAlmostEqual(actual_sale_order.amount_total, expected_amount_total,
+                               msg=f'Sale order amount total: {msg} {order_qty_msg}',
+                               delta=self.precision_delta * qty)  # because price_unit is rounded then mult by qty
+        amount_total_computed_with_price_reduce = self.event_ticket.with_context({
+            'pricelist': self.price_list.id,
+            'quantity': qty}).price_reduce_taxinc * qty
+        self.assertAlmostEqual(amount_total_computed_with_price_reduce, expected_amount_total,
+                               msg=f'Amount total computed with price reduce tax incl: {msg} {order_qty_msg}',
+                               delta=self.precision_delta * qty)
+
+    def test_ticket_price_with_price_list_extensive(self):
+        # Only vary price list currency because product and event currency are related to company.currency (see models)
+        for currency in (self.usd_currency, self.eur_currency):
+            for use_product_tpl in (False, True):
```

Comment:
Seems worth it to build `cases` as in the other file to gain an indentation level. Or use
```suggestion
        for currency, use_product_tpl in itertools.product(
                (self.usd_currency, self.eur_currency), (False, True)):
```

Reply:
yes, thanks for the hint, I have used itertools as it is more compact